tests/jenkins/TestBenchmarkPullRequest.groovy

```diff
@@ -0,0 +1,198 @@
+        binding.setVariable('HAS_SECURITY', 'false')
+        binding.setVariable('SINGLE_NODE_CLUSTER', 'true')
+        binding.setVariable('MIN_DISTRIBUTION', 'true')
+        binding.setVariable('USE_50_PERCENT_HEAP', 'true')
+        binding.setVariable('SUFFIX', '1234')
+        binding.setVariable('MANAGER_NODE_COUNT', '')
+        binding.setVariable('DATA_NODE_COUNT', '')
+        binding.setVariable('ENABLE_REMOTE_STORE', 'false')
+        binding.setVariable('USER_TAGS', 'run-type:test')
+        binding.setVariable('WORKLOAD_PARAMS', '')
+        binding.setVariable('TEST_PROCEDURE', 'append-no-conflicts')
+        binding.setVariable('EXCLUDE_TASKS', '')
+        binding.setVariable('INCLUDE_TASKS', '')
+        binding.setVariable('ADDITIONAL_CONFIG', '')
+        binding.setVariable('CLIENT_NODE_COUNT', '')
+        binding.setVariable('INGEST_NODE_COUNT', '')
+        binding.setVariable('ML_NODE_COUNT', '')
+        binding.setVariable('DATA_NODE_STORAGE', '100')
+        binding.setVariable('ML_NODE_STORAGE', '')
+        binding.setVariable('DATA_INSTANCE_TYPE', 'r5-4xlarge')
+        binding.setVariable('JVM_SYS_PROPS', '')
+        binding.setVariable('CAPTURE_NODE_STAT', 'false')
+        binding.setVariable('CAPTURE_SEGMENT_REPLICATION_STAT', 'false')
+        binding.setVariable('JOB_NAME', 'benchmark-test')
+        binding.setVariable('BENCHMARK_TEST_CONFIG_LOCATION', 'test_config')
+        binding.setVariable('PUBLIC_ARTIFACT_URL', 'test://artifact.url')
+        binding.setVariable('STAGE_NAME', 'test_stage')
+        binding.setVariable('TEST_WORKLOAD', 'nyc-taxis')
+        binding.setVariable('WEBHOOK_URL', 'test://artifact.url')
+        binding.setVariable('TELEMETRY_PARAMS', '')
+        binding.setVariable('pull_request_number', '1234')
+        binding.setVariable('pull_request', 1234)
+        binding.setVariable('repository','opensearch-project/OpenSearch')
+
+        helper.registerAllowedMethod("GenericTrigger", [Map], { println 'GenericTrigger called with params: ' + it })
+    }
+
+    @Test
+    public void testBenchmarkPullRequestGenericCause_verifyPipeline() {
+        binding.getVariable('currentBuild').rawBuild = [:]
+        binding.getVariable('currentBuild').rawBuild.getCauses = { return "jenkins.branch.GenericCause@123abc" }
+
+        super.testPipeline("jenkins/opensearch/benchmark-pull-request.jenkinsfile",
+                "tests/jenkins/jenkinsjob-regression-files/opensearch/benchmark-pull-request-generic.jenkinsfile")
+    }
+
+    @Test
+    public void testBenchmarkPullRequestUserIdCause_verifyPipeline() {
+        binding.getVariable('currentBuild').rawBuild = [:]
+        binding.getVariable('currentBuild').rawBuild.getCauses = { return "jenkins.branch.UserIdCause@123abc" }
+
+        super.testPipeline("jenkins/opensearch/benchmark-pull-request.jenkinsfile",
+                "tests/jenkins/jenkinsjob-regression-files/opensearch/benchmark-pull-request-user.jenkinsfile")
```

Comment:
I dont think this test is required. It is not asserting anything so we can remove this and let next test take care of it

Reply:
This is just to make sure the job works as expected for different types to triggers.